extensions/replication/replicationStatusProcessor/ReplicationStatusProcessor.js

```diff
@@ -124,6 +126,42 @@ class ReplicationStatusProcessor {
         return this._consumer.close(done);
     }
 
+    /**
+     * Set the Redis hash key for each failed backend.
+     * @param {QueueEntry} queueEntry - The queue entry with the failed status.
+     * @param {Object} kafkaEntry - The kafka entry with the failed status.
+     * @param {Function} cb          [description]
+     * @return {undefined}
+     */
+    _setFailedKeys(queueEntry, kafkaEntry, cb) {
+        const { status, backends } = queueEntry.getReplicationInfo();
+        if (status !== 'FAILED') {
+            return process.nextTick(() => cb());
+        }
+        const bucket = queueEntry.getBucket();
+        const key = queueEntry.getObjectKey();
+        const versionId = queueEntry.getEncodedVersionId();
+        const fields = [];
+        backends.forEach(backend => {
+            const { status, site } = backend;
+            if (status !== 'FAILED') {
+                return undefined;
+            }
+            const field = `${bucket}:${key}:${versionId}:${site}`;
```

Comment:
I decided to use "field" because it corresponds to the Redis naming scheme where the "key" is the name of the Redis key, in this case a hash data type with multiple "fields". https://redis.io/commands/hmset. I can leave a comment for clarity on this.

Reply:
ok thanks for the explanation!